ddp-workspace/projects/ddp-sdk/src/lib/components/activityForm/picklist/autocompleteActivityPicklistQuestion.component.ts

```diff
@@ -0,0 +1,148 @@
+                <span [innerHtml]="suggestion.label | searchHighlight: autocompleteInput.value"></span>
+            </mat-option>
+        </mat-autocomplete>
+    </mat-form-field>`,
+    styles: [`
+        .width {
+            width: 100%;
+        }
+
+        .ddp-autocomplete-field ::ng-deep .mat-form-field-infix {
+          width: auto;
+        }
+
+        .parent {
+            font-weight: bold;
+        }
+
+        .child ::ng-deep .mat-option-text {
+            padding-left: 13px;
+        }
+    `]
+})
+export class AutocompleteActivityPicklistQuestion extends BaseActivityPicklistQuestion implements OnInit, OnDestroy {
+    private allSuggestions: Suggestion[] = [];
+    private readonly ngUnsubscribe = new Subject();
+
+    filteredSuggestions: Suggestion[];
+    inputFormControl = new FormControl();
+
+    constructor(translate: NGXTranslateService) {
+        super(translate);
+    }
+
+    public ngOnInit(): void {
+        this.allSuggestions = this.generateSuggestions();
+        const customVersionId = this.block.picklistOptions.find(option => option.allowDetails)?.stableId;
+
+        this.inputFormControl.valueChanges.pipe(
+            map(value => typeof value === 'string' ? value.trim() : value),
+            distinctUntilChanged(),
+            debounceTime(200),
+            takeUntil(this.ngUnsubscribe)
+        ).subscribe((value: string | Suggestion) => {
+            const query = typeof value === 'string' ? value : value.label;
+            this.filteredSuggestions = query ? this.filterOptions(query) : this.allSuggestions.slice();
+
+            if (typeof value === 'string') {
+                const minimalLengthForOption = 2;
+                if (value.length >= minimalLengthForOption) {
+                    const matchInSuggestions = this.filteredSuggestions
+                        .find(suggestion => suggestion.label.toLowerCase() === value.toLowerCase());
+                    if (matchInSuggestions) {
+                        this.valueChanged.emit([{ stableId: matchInSuggestions.value, detail: null }]);
+                    } else if (customVersionId) {
+                        this.valueChanged.emit([{ stableId: customVersionId, detail: value }]);
+                    }
+                }
+            } else {
+                this.valueChanged.emit([{ stableId: value.value, detail: null }]);
+            }
+        });
+    }
+
+    public ngOnDestroy(): void {
+        super.ngOnDestroy();
+        this.ngUnsubscribe.next();
+        this.ngUnsubscribe.complete();
+    }
+
+    private generateSuggestions(): Suggestion[] {
+        return this.block.picklistOptions.reduce((acc, currentValue) => {
+            if (currentValue.allowDetails) return acc;
+            const suggestion: Suggestion = {
+                label: currentValue.optionLabel,
+                value: currentValue.stableId,
+            };
+            acc.push(suggestion);
+            if (currentValue.nestedOptions) {
```

Comment:
Consider checking explicitly for empty array (here, empty array would evaluate to true), so would be treated as a parent even though no children.

Reply:
good idea! done